projects/angular/src/button/providers/button-group-focus-handler.service.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ChangeDetectorRef, Injectable } from '@angular/core';
+
+import { FocusService } from '../../utils/focus/focus.service';
+import { FocusableItem } from '../../utils/focus/focusable-item/focusable-item';
+import { Linkers } from '../../utils/focus/focusable-item/linkers';
+import { uniqueIdFactory } from '../../utils/id-generator/id-generator.service';
+
+@Injectable()
+export class ButtonGroupFocusHandler {
+  constructor(private focusService: FocusService) {}
+
+  // We need a Change Detector from the related component, so we can update it on Blur
+  // (which is needed because of Edge specific lifecycle mis-behavior)
+  componentCdRef: ChangeDetectorRef;
```

Comment:
Since this class is provided at the the component injector, the `ChangeDirectorRef` can be injected directly into this class.